client/add/add.ctrl.js

```diff
@@ -123,11 +127,18 @@ module.exports = class Add {
            toastAction,
            toastAction ? `/pokemon/${successfulUploads[0].created.id}` : undefined);
       })
-      .filter(line => line.success && line.created.box === box.data.id)
-      .then(lines => lines.slice(0, BOX_PAGE_SIZE - (box.data.contents.length % BOX_PAGE_SIZE)))
+      .filter(line => line.success)
       .map(line => line.created)
-      .then(lines => box.data.contents.push(...lines))
-      .tap(box.onscroll)
+      .each(line => {
+        if (this.boxesById[line.box].contents.length % BOX_PAGE_SIZE) {
+          this.boxesById[line.box].contents.push(line);
+        }
+        if (selectedBox && line.box === selectedBox.id
+          && selectedBox.data.contents.length % BOX_PAGE_SIZE) {
+          selectedBox.data.contents.push(line);
```

Comment:
Isn't this redundant with the `if` statement above it? It looks like it will cause things to get added to the selected box twice.